packages/@aws-cdk/aws-vpcv2-alpha/lib/ipam.ts

```diff
@@ -0,0 +1,428 @@
+
+/**
+ * Creates new IPAM Pool Cidr for IPV6
+ * Required to assoicate IP address for Public IP source set to Amazon Owned IPv6 Address
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampoolcidr.html
+ * @resource AWS::EC2::IPAMPoolCidr
+ */
+export class IpamPoolCidr extends Resource {
+
+  constructor(scope: Construct, id: string, props: IpamPoolCidrProps) {
+    super(scope, id, {
+      physicalName: props.ipamPoolCidrName ?? Lazy.string({
+        produce: () => Names.uniqueResourceName(this, { maxLength: 128, allowedSpecialCharacters: '_' }),
+      }),
+    });
+    this.node.defaultChild = new CfnIPAMPoolCidr(this, 'PoolCidr', {
+      netmaskLength: props.netmaskLength,
+      ipamPoolId: props.ipamPoolId,
+      cidr: props.cidr,
+    });
+  }
+}
+
+/**
+ * Creates custom Ipam Scope, can only be private
+ * (can be used for adding custom scopes to an existing IPAM)
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipamscope.html
+ * @resource AWS::EC2::IPAMScope
+ */
+export class IpamScope extends Resource {
+
+  private readonly _ipamScope: CfnIPAMScope;
+
+  /**
+   * ID for Resource IpamScope
+   * @attribute IpamScopeId
+   */
+  public readonly ipamScopeId: string;
+  constructor(scope: Construct, id: string, props: IpamScopeProps) {
+    super(scope, id);
+    this._ipamScope = new CfnIPAMScope(scope, 'IpamScope', {
+      ipamId: props.ipamId,
+    });
+    this.ipamScopeId = this._ipamScope.attrIpamScopeId;
+  }
+}
+
+/**
+ * Base class for IPAM scopes.
+ */
+abstract class IpamScopeBase {
+  protected constructor(
+    protected readonly scope: Construct,
+    protected readonly scopeId: string,
+  ) {}
+
+  /**
+   * Adds a pool to the IPAM scope.
+   */
+  addPool(id: string, options: PoolOptions): IpamPool {
+    const pool = new IpamPool(this.scope, id, {
+      ipamPoolName: id,
+      addressFamily: options.addressFamily,
+      provisionedCidrs: options.provisionedCidrs,
+      ipamScopeId: this.scopeId,
+      locale: options.locale ?? Stack.of(this.scope).region,
+      publicIpSource: options.publicIpSource,
+      awsService: options.awsService,
+    });
+
+    return pool;
+  }
+}
+
+/**
+ * Provides access to default public IPAM scope through add pool method.
+ * Usage: To add an Ipam Pool to a default public scope
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipamscope.html
+ */
+export class IpamPublicScope extends IpamScopeBase {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+  }
+}
+
+/**
+ * Provides access to default private IPAM scope through add pool method.
+ * Usage: To add an Ipam Pool to a default private scope
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipamscope.html
+ */
+export class IpamPrivateScope extends IpamScopeBase {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+  }
+}
```

Comment:
i think we need this since both differs in terms of user DX, but will discuss for overall improvements that you see we should implement here

```
const pool1 = ipam.privateScope.addPool('PrivatePool0', {
  addressFamily: AddressFamily.IP_V4,
  provisionedCidrs: [{ cidr: '10.2.0.0/16' }],
  locale: 'eu-central-1',
});
```